### PR TITLE
Fix goto implementation does not suggest all subtypes

### LINKED ISSUE #29724: Goto Implementation should suggest all subtypes
Steps to reproduce:

```typescript
interface Person {
}

class Teacher implements Person {
}

class SeniorTeacher extends Teacher {
}
```

Goto implementation on `Person` only lists `Teacher`. It should list `SeniorTeacher` as well.

### PR DESCRIPTION
**Pull request by fuafa**
Fixes #29724

### CODE REVIEW

File: src/services/findAllReferences.ts [hunk 1]
```diff
@@ -203,7 +203,40 @@ namespace ts.FindAllReferences {
...
+        else {
+            const queue = entries && [...entries];
+            const seenNodes = createMap<true>();
+            while (queue && queue.length) {
+                const entry = queue.shift() as NodeEntry;
+
+                const key = `${entry.node.getSourceFile().fileName}${entry.node.pos}`;
+                if (seenNodes.has(key)) {
+                    continue;
+                }
```
Comment (andrewbranch) [inline src/services/findAllReferences.ts:226]:
```suggestion
                if (!addToSeen(getNodeId(entry.node))) {
                    continue;
                }
```

File: src/services/findAllReferences.ts [hunk 2]
```diff
@@ -203,7 +203,40 @@ namespace ts.FindAllReferences {
...
+                    referenceEntries.push(entry);
+                }
+                else {
+                    referenceEntries = [entry];
+                }
+                const entries = getImplementationReferenceEntries(program, cancellationToken, sourceFiles, entry.node, entry.node.pos);
+                if (entries) {
+                    queue.push(...entries);
+                }
+                seenNodes.set(key, true);
```
Comment (andrewbranch) [inline src/services/findAllReferences.ts:237]: (Handled by `addToSeen` above)

```suggestion
```

File: src/services/findAllReferences.ts [hunk 3]
```diff
@@ -203,7 +203,40 @@ namespace ts.FindAllReferences {
...
+                const key = `${entry.node.getSourceFile().fileName}${entry.node.pos}`;
+                if (seenNodes.has(key)) {
+                    continue;
+                }
+                if (referenceEntries) {
+                    referenceEntries.push(entry);
+                }
+                else {
+                    referenceEntries = [entry];
+                }
```
[the remark this answers is not in the document]
Comment (fuafa) [reply src/services/findAllReferences.ts]: fixed

File: src/services/findAllReferences.ts (hunk 1 above)
Comment (fuafa) [reply src/services/findAllReferences.ts]: fixed

File: src/services/findAllReferences.ts (hunk 2 above)
Comment (fuafa) [reply src/services/findAllReferences.ts]: fixed

**Pull request merged**